packages/themes/src/createTheme.ts

```diff
@@ -1,20 +1,317 @@
+ * @param darkModeSelector - Dark mode selector string, or null to opt out of CSS generation
+ */
+export function convertTuplesToCssVariables(
+  variables: Record<string, string | [string, string] | number | undefined>,
+  darkModeSelector?: string | null,
+): ProcessedVariables {
+  const lightRules: string[] = [];
+  const darkRules: string[] = [];
+  const processedVariables: Record<string, string | number | undefined> = {};
+
+  for (const [key, value] of Object.entries(variables)) {
+    if (Array.isArray(value) && value.length === 2) {
+      const [lightValue, darkValue] = value;
+
+      if (darkModeSelector === null) {
+        processedVariables[key] = lightValue;
+      } else {
+        const cssVarName = `${CSS_VAR_PREFIX}${toKebabCase(key)}`;
+
+        lightRules.push(`    ${cssVarName}: ${lightValue};`);
+        darkRules.push(`    ${cssVarName}: ${darkValue};`);
+
+        processedVariables[key] = `var(${cssVarName})`;
+      }
+    } else {
+      processedVariables[key] = value;
+    }
+  }
+
+  let cssString: string | null = null;
+  if (lightRules.length > 0 && darkModeSelector !== null) {
+    const normalizedSelector = darkModeSelector as string;
+
+    if (normalizedSelector.startsWith(MEDIA_QUERY_PREFIX)) {
+      // Media query - wrap dark rules in :root
+      cssString = `
+    :root {
+${lightRules.join('\n')}
+    }
+    ${normalizedSelector} {
+      :root {
+${darkRules.join('\n')}
+      }
+    }
+  `.trim();
+    } else {
+      // Regular selector - use as-is
+      cssString = `
+    :root {
+${lightRules.join('\n')}
+    }
+    ${normalizedSelector} {
+${darkRules.join('\n')}
+    }
+  `.trim();
+    }
+  }
+
+  return { cssString, processedVariables };
+}
+
+/**
+ * Transforms a CSS selector for proper nesting context
+ * Class selectors like '.dark' become '.dark &' for parent-child relationship
+ * ID selectors like '#theme' become '#theme &' for parent-child relationship
+ */
+export function transformSelectorForNesting(selector: string): string {
+  // If it's a class or ID selector, append & for proper CSS nesting
+  if (
+    (selector.startsWith(CLASS_SELECTOR_PREFIX) || selector.startsWith(ID_SELECTOR_PREFIX)) &&
+    !selector.includes('&')
+  ) {
+    return `${selector} &`;
+  }
+  // For media queries and other selectors, return as-is
+  return selector;
+}
+
+/**
+ * Resolves elements configuration - handles both static elements and function-based elements
+ */
+export function resolveElementsConfiguration(
+  elements: Elements | DarkModeElementsFunction | undefined,
+  darkModeSelector: string | null,
+): Elements | undefined {
+  if (!elements) {
+    return undefined;
+  }
+
+  // If it's a function, call it with darkModeSelector
+  if (typeof elements === 'function') {
+    // If opted out of dark mode, don't call the function at all
+    // since it's designed to generate dark mode styles
+    if (darkModeSelector === null) {
+      return undefined;
+    }
```

Comment:
This seems unexpected to me. If no dark mode selector is provided, but `elements` is a function, we ignore any elements styling? Should we warn or error in this case if the function syntax requires a `darkModeSelector`?